core/trino-main/src/test/java/io/trino/connector/system/TestMaterializedViewSystemTable.java

```diff
@@ -0,0 +1,134 @@
+    private static final String SELECT_SOURCE_TABLE_SQL = "SELECT col1, col2, col3 FROM test_table";
+
+    private static final QualifiedObjectName FULLY_QUALIFIED_NAME_1 = new QualifiedObjectName(
+            "test_catalog_name_1",
+            "test_schema_name_1",
+            "test_schema_name_1");
+    private static final QualifiedObjectName FULLY_QUALIFIED_NAME_2 = new QualifiedObjectName(
+            "test_catalog_name_2",
+            "test_schema_name_2",
+            "test_schema_name_2");
+    private static final CatalogSchemaTableName STORAGE_TABLE_NAME = new CatalogSchemaTableName(
+            "test_catalog",
+            "test_schema",
+            "test_table");
+
+    private MaterializedViewSystemTable systemTable;
+
+    @BeforeClass
+    public void setup()
+    {
+        systemTable = new MaterializedViewSystemTable(dummyMetadata(), new AllowAllAccessControl());
+    }
+
+    @DataProvider
+    public Object[][] getTestMaterializedViews()
+    {
+        return new Object[][] {
+                {
+                        FULLY_QUALIFIED_NAME_1,
+                        createMaterializedViewDefinition(FULLY_QUALIFIED_NAME_1, Optional.of(MATERIALIZED_VIEW_COMMENT_1)),
+                        new MaterializedViewFreshness(false),
+                        createMaterializedViewRow(FULLY_QUALIFIED_NAME_1, Optional.of(MATERIALIZED_VIEW_COMMENT_1), false)
+                },
+                {
+                        FULLY_QUALIFIED_NAME_2,
+                        createMaterializedViewDefinition(FULLY_QUALIFIED_NAME_2, Optional.of(MATERIALIZED_VIEW_COMMENT_2)),
+                        new MaterializedViewFreshness(true),
+                        createMaterializedViewRow(FULLY_QUALIFIED_NAME_2, Optional.of(MATERIALIZED_VIEW_COMMENT_2), true)
+                },
+                {
+                        FULLY_QUALIFIED_NAME_2,
+                        createMaterializedViewDefinition(FULLY_QUALIFIED_NAME_2, Optional.empty()),
+                        new MaterializedViewFreshness(true),
+                        createMaterializedViewRow(FULLY_QUALIFIED_NAME_2, Optional.empty(), true)
+                },
+                {
+                        FULLY_QUALIFIED_NAME_2,
+                        createMaterializedViewDefinition(FULLY_QUALIFIED_NAME_2, Optional.of(EMPTY_COMMENT)),
+                        new MaterializedViewFreshness(true),
+                        createMaterializedViewRow(FULLY_QUALIFIED_NAME_2, Optional.of(EMPTY_COMMENT), true)
+                }
+        };
+    }
+
+    @Test(dataProvider = "getTestMaterializedViews")
+    public void toTableRows(
+            QualifiedObjectName name,
+            ConnectorMaterializedViewDefinition definition,
+            MaterializedViewFreshness freshness,
+            Object[] materializedViewRow)
+    {
+        assertThat(systemTable.createMaterializedViewRow(name, freshness, definition))
```

Comment:
I don't think we're gaining by just testing this method in isolation now. The connector tests should cover all the scenarios. We can remove this class.